src/core/init/manifest_update_scheduler.ts

```diff
@@ -20,103 +20,159 @@ import {
+  fetchManifest : IManifestFetcher;
+  initialManifest : { manifest : Manifest;
+                      sendingTime? : number; };
+  manifestUpdateUrl : string | undefined;
+  maximumManifestUpdateInterval? : number;
+  minimumManifestUpdateInterval : number;
+  scheduleRefresh$ : Observable<{ completeRefresh : boolean }>;
+}
+
 /**
  * Refresh the Manifest at the right time.
- * @param {Object} initialManifest
- * @param {Observable} scheduleRefresh$
- * @param {Function} fetchManifest
- * @param {number} minimumManifestUpdateInterval
+ * @param {Object} manifestUpdateSchedulerArguments
  * @returns {Observable}
  */
-export default function manifestUpdateScheduler(
-  initialManifest : { manifest : Manifest;
-                      sendingTime? : number; },
-  scheduleRefresh$ : Observable<number>,
-  fetchManifest : IManifestFetcher,
-  minimumManifestUpdateInterval : number
-) : Observable<never> {
+export default function manifestUpdateScheduler({
+  fetchManifest,
+  initialManifest,
+  manifestUpdateUrl,
+  maximumManifestUpdateInterval,
+  minimumManifestUpdateInterval,
+  scheduleRefresh$,
+} : IManifestUpdateSchedulerArguments) : Observable<never> {
+  // The Manifest always keeps the same Manifest
+  const { manifest } = initialManifest;
+
   function handleManifestRefresh$(
     manifestInfos: { manifest: Manifest;
                      sendingTime?: number; }): Observable<never> {
-    const { manifest, sendingTime } = manifestInfos;
-    // schedule a Manifest refresh to avoid sending too much request.
-    const manualRefresh$ = scheduleRefresh$.pipe(
-      mergeMap((delay) => {
-        const timeSinceLastRefresh = sendingTime == null ?
-                                       0 :
-                                       performance.now() - sendingTime;
-        const minInterval = Math.max(minimumManifestUpdateInterval
-                                       - timeSinceLastRefresh,
-                                     0);
-        return observableTimer(Math.max(delay - timeSinceLastRefresh,
-                                        minInterval));
-      }));
-
-    const autoRefresh$ = (() => {
-      if (manifest.lifetime == null || manifest.lifetime <= 0) {
-        return EMPTY;
-      }
-      const timeSinceRequest = sendingTime == null ?
-                                 0 :
-                                 performance.now() - sendingTime;
-      const minInterval = Math.max(minimumManifestUpdateInterval
-                                     - timeSinceRequest,
-                                   0);
-      const updateTimeout = manifest.lifetime * 1000 - timeSinceRequest;
-      return observableTimer(Math.max(updateTimeout, minInterval));
-    })();
-
-    const expired$ = manifest.expired === null ? EMPTY :
-                                                 observableFrom(manifest.expired);
+    const { sendingTime } = manifestInfos;
+
+    const manualRefresh$ = scheduleRefresh$.pipe(mergeMap(({ completeRefresh }) =>
```

Comment:
Is `manualRefresh` really adaptated ? What makes it manual, if it is the RxPlayer that ask for a refresh ?

Reply:
I'm renaming it to `internalRefresh$`

Reply:
Done